oxidation/libparsec/crates/crypto/src/sodiumoxide/secret.rs

```diff
@@ -72,13 +66,30 @@ impl SecretKey {
     }
 }
 
+crate::macros::impl_key_debug!(SecretKey);
+
 impl AsRef<[u8]> for SecretKey {
     #[inline]
     fn as_ref(&self) -> &[u8] {
         self.0.as_ref()
     }
 }
 
+impl TryFrom<&[u8]> for SecretKey {
+    type Error = CryptoError;
+
+    fn try_from(v: &[u8]) -> Result<Self, Self::Error> {
+        let arr: [u8; Self::SIZE] = v.try_into().map_err(|_| CryptoError::DataSize)?;
+        Ok(Self(Key(arr)))
```

Comment:
```suggestion
        let arr: [u8; KEYBYTES] = v.try_into().map_err(|_| CryptoError::DataSize)?;
        Ok(Self::from(arr))
```

Re-use the `From<[u8; KEYBYTES]>`

Reply:
Done